substrate/frame/asset-conversion/ops/src/lib.rs

```diff
@@ -0,0 +1,325 @@
+						Preservation::Expendable,
+					)?,
+				Error::<T>::PartialTransfer
+			);
+
+			ensure!(
+				balance3 ==
+					T::PoolAssets::transfer(
+						info.lp_token.clone(),
+						&prior_account,
+						&new_account,
+						balance3,
+						Preservation::Expendable,
+					)?,
+				Error::<T>::PartialTransfer
+			);
+
+			// Refund deposits from prior accounts and burn previously minted assets.
+
+			if let Some((d, b)) =
+				T::AssetsRefund::deposit_held(asset1.clone(), prior_account.clone())
+			{
+				T::AssetsRefund::refund(asset1.clone(), prior_account.clone())?;
+				let ed = T::DepositAsset::minimum_balance();
+				T::DepositAsset::burn_from(&d, b + ed, Precision::Exact, Fortitude::Force)?;
+			}
+
+			if let Some((d, b)) =
+				T::AssetsRefund::deposit_held(asset2.clone(), prior_account.clone())
+			{
+				T::AssetsRefund::refund(asset2.clone(), prior_account.clone())?;
+				let ed = T::DepositAsset::minimum_balance();
+				T::DepositAsset::burn_from(&d, b + ed, Precision::Exact, Fortitude::Force)?;
+			}
+
+			if let Some((d, b)) =
+				T::PoolAssetsRefund::deposit_held(info.lp_token.clone(), prior_account.clone())
+			{
+				T::PoolAssetsRefund::refund(info.lp_token.clone(), prior_account.clone())?;
+				let ed = T::DepositAsset::minimum_balance();
+				T::DepositAsset::burn_from(&d, b + ed, Precision::Exact, Fortitude::Force)?;
+			}
+
+			T::PoolAssetsTeam::reset_team(
+				info.lp_token,
+				new_account.clone(),
+				new_account.clone(),
+				new_account.clone(),
+				new_account.clone(),
+			)?;
+
+			Self::deposit_event(Event::MigratedToNewAccount {
+				pool_id,
+				prior_account,
+				new_account,
+			});
+
+			Ok(Pays::No.into())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		/// Returns the prior and new account IDs for a given pool ID. The prior account ID comes
+		/// first in the tuple.
+		#[cfg(not(any(test, feature = "runtime-benchmarks")))]
+		fn addresses(pool_id: &T::PoolId) -> Option<(T::AccountId, T::AccountId)> {
+			match (
+				T::PriorAccountIdConverter::try_convert(pool_id),
+				T::PoolLocator::address(pool_id),
+			) {
+				(Ok(a), Ok(b)) if a != b => Some((a, b)),
+				_ => None,
+			}
+		}
+
+		/// Returns the prior and new account IDs for a given pool ID. The prior account ID comes
+		/// first in the tuple.
+		///
+		/// This function is intended for use only in test and benchmark environments. The prior
+		/// account ID represents the new account ID from [`Config::PoolLocator`], allowing the use
+		/// of the main pallet's calls to set up a pool with liquidity placed in that account and
+		/// migrate it to another account, which in this case is the result of
+		/// [`Config::PriorAccountIdConverter`].
+		#[cfg(any(test, feature = "runtime-benchmarks"))]
+		pub(crate) fn addresses(pool_id: &T::PoolId) -> Option<(T::AccountId, T::AccountId)> {
+			match (
+				T::PoolLocator::address(pool_id),
+				T::PriorAccountIdConverter::try_convert(pool_id),
+			) {
+				(Ok(a), Ok(b)) if a != b => Some((a, b)),
```

Comment:
Instead of these 2 separate functions, I'd just change the actual return here and update the doc comment on the original function accordingly.

```suggestion
				(Ok(a), Ok(b)) if a != b => {
				   #[cfg(not(any(test, feature = "runtime-benchmarks")))]
				   Some((a, b))
				   /// Explain why
				   #[cfg(any(test, feature = "runtime-benchmarks"))]
				   Some((b, a))
				},
```

Reply:
I like separate function. It also a more common style within our crates.